From f5b4b79dcf6323731cae765e06c6b58fea0ce747 Mon Sep 17 00:00:00 2001
From: Eli Schwartz <eschwartz93@gmail.com>
Date: Sun, 31 Jul 2022 13:23:22 -0400
Subject: [PATCH] meson: fix types for some build options

Booleans are supposed to be actual booleans, not strings describing
their value, but Meson has a bug where it silently accepts either one.
It's still wrong to do it though, and other implementations of Meson
such as muon choke on it.

Signed-off-by: Eli Schwartz <eschwartz93@gmail.com>
---
 meson_options.txt | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/meson_options.txt b/meson_options.txt
index 16377dcc34..ed3b116778 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -61,12 +61,12 @@ option('listen_local', type: 'boolean', value: true,
 option('int10', type: 'combo', choices: ['stub', 'x86emu', 'vm86', 'auto', 'false'],
        value: 'auto',
        description: 'Xorg int10 backend (default: usually x86emu)')
-option('suid_wrapper', type: 'boolean', value: 'false',
+option('suid_wrapper', type: 'boolean', value: false,
        description: 'SUID wrapper for legacy driver support')
-option('pciaccess', type: 'boolean', value: 'true',
+option('pciaccess', type: 'boolean', value: true,
        description: 'Xorg pciaccess support')
-option('udev', type: 'boolean', value: 'true')
-option('udev_kms', type: 'boolean', value: 'true')
+option('udev', type: 'boolean', value: true)
+option('udev_kms', type: 'boolean', value: true)
 option('hal', type: 'combo', choices: ['true', 'false', 'auto'], value: 'auto',
        description: 'Enable HAL integration')
 option('systemd_logind', type: 'combo', choices: ['true', 'false', 'auto'], value: 'auto',
@@ -111,7 +111,7 @@ option('xf86-input-inputtest', type: 'boolean', value: true,
 option('dri1', type: 'combo', choices: ['true', 'false', 'auto'], value: 'auto', description: 'Build DRI1 extension (default: auto)')
 option('dri2', type: 'combo', choices: ['true', 'false', 'auto'], value: 'auto', description: 'Build DRI2 extension (default: auto)')
 option('dri3', type: 'combo', choices: ['true', 'false', 'auto'], value: 'auto', description: 'Build DRI3 extension (default: auto)')
-option('drm', type: 'boolean', value: 'true',
+option('drm', type: 'boolean', value: true,
         description: 'Build Xorg with libdrm support')
 
 option('apple-applications-dir', type: 'string', value: '/Applications/Utilities',
-- 
GitLab

